src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```
@@ -1209,6 +1229,10 @@ internal static Task LogDowngradeWarningsOrErrorsAsync(IEnumerable<RestoreTarget
                 {
                     failed = true;
                 }
+                catch (AggregateException e) when (e.InnerException is FatalProtocolException)
+                {
+                    failed = true;
```

This is fine if there's only a single exception wrapped in the AggregateException, but if there are multiple exceptions, and only the first one is a `FatalProtocolException`, shouldn't we do something?  I'm guessing FatalProtocolException was special cased, because it gets logged somewhere, but if we have a bug causing a NullReferenceException, ArgumentException, or similar, it will be very hard to isolate if we silently ignore the exception.

Perhaps change it to something like `when (e.InnerExceptions.All(e => e is FatalogProcoolException))`